thoth/storages/graph/postgres.py

```diff
@@ -1308,172 +1590,871 @@ def get_unsolved_python_package_versions_count_all(
         if distinct:
             query = query.distinct()
 
-        query = query.count()
+        result = query.count()
 
-        unsolved = self._get_unsolved_python_package_edge_cases(
-            package_name=package_name,
-            package_version=package_version
-        )
+        unsolved = []
+        if len(result) < count:
+            unsolved = self._get_unsolved_python_package_versions_edge_cases(
+                package_name=package_name,
+                package_version=package_version,
+                distinct=distinct,
+                is_count=True
+            )
 
-        total_count = query + len(unsolved)
+        total_count = result + len(unsolved)
```

Comment:
For performance, it would be good to introduce a parameter to `_get_unsolved_python_package_versions_edge_cases` which will trigger counting (its a private method, so it should be ok).